clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java

```diff
@@ -1481,6 +1481,7 @@ private void parseLegacyJDBCStore(XMLExtendedStreamReader reader, PathAddress ca
                     break;
                 }
                 default: {
+                    address = address != null ? address : cacheAddress;
```

Comment:
This is not correct.  If address is null here, then we'd actually want to throw an exception, since either entry-table, bucket-table, or both; must be defined previously (which determines the target address).

Reply:
@pferraro Thanks for your feedback. I would suggest [o.w.common.Assert.impossibleSwitchCase](https://github.com/wildfly/wildfly-common/blob/master/src/main/java/org/wildfly/common/Assert.java#L552) or do you prefer another (new) one?

Reply:
In this case, a null address is unexpected, but not "impossible" - and indicates that the xml violates the schema.  However, since we do not formally validate against our schema before parsing, this is a condition that we should handle during parsing. I suggest using:
```java
if (address == null) {
    throw ParseUtils.missingOneOf(reader, EnumSet.of(XMLElement.ENTRY_TABLE, XMLElement.BUCKET_TABLE));
}
```